b2/console_tool.py

```diff
@@ -355,6 +355,100 @@ def run(self, args):
         return 0
 
 
+class CopyFile(Command):
+    """
+    b2 copy-file [--metadataDirective [copy|replace]] [--contentType <contentType>] \\
+                 [--info <key>=<value>]* [--range start,end] \\
+                 <sourceFileId> <destinationBucketName> <b2FileName>
```

Comment:
Do we want to a copy-by-name operation?

Reply:
It would not be atomic. We can always easily add it later if requested, but I see value in the user retrieving the new file_id and then passing it here (in 2 lines of bash), as then the user and not b2cli, is responsible for dealing with concurrency. Also generally I think the amount of non-essential commands in cli should be kept to minimum, or we'll one day find ourselves in a situation where cli has too many commands for the user to comprehend.

Reply:
I guess I could make the same comment about the server-side API: specifying the source by name would be nice.  Since we don't do that, I think this interface is fine.

Do we want to leave room in the interface for names in the future.  We could have a `--sourceFileId` option, and then later add `--sourceFileName` and `--sourceBucket` options without breaking compatibility.  (Not sure that's a good idea.  Just thinking out loud.)

Reply:
if we do, then we should be consistent with downloads
```
b2 download-file-by-id [--noProgress] <fileId> <localFileName>
b2 download-file-by-name [--noProgress] <bucketName> <fileName> <localFileName>
```
so `copy-file-by-id` (and in the future maybe `copy-file-by-name` too (now that I think about it, that is a very small feature).

I'm convinced. Shall I change the name?

Reply:
Yes.  Looks good to me.